cpp-package/src/inc/brainflow_array.h

```diff
@@ -0,0 +1,468 @@
+#pragma once
+
+#include <algorithm>
+#include <array>
+#include <iostream>
+#include <memory>
+#include <stdlib.h>
+#include <string.h>
+
+#include "brainflow_constants.h"
+#include "brainflow_exception.h"
+
+
+template <typename T, size_t Dim>
+class BrainFlowArray;
+template <typename T, size_t Dim>
+std::ostream &operator<< (std::ostream &os, const BrainFlowArray<T, Dim> &array);
+
+template <size_t N>
+static inline int product (const std::array<int, N> &array)
+{
+    int result = 1;
+
+    for (int i = 0; i < N; i++)
+    {
+        result *= array[i];
+    }
+
+    return result;
+}
+
+template <size_t N>
+static inline std::array<int, N> zero_array ()
+{
+    std::array<int, N> result;
+
+    for (int i = 0; i < N; i++)
+    {
+        result[i] = 0;
+    }
+
+    return std::move (result);
+}
+
+static inline std::array<int, 1> make_array (int size0)
```

Comment:
The parameters here are named `size`, but the resulting arrays are holding them as values, not sizes.  Not sure if that is intentional.  These are consolidatable into one template function, not sure if it is your style to do so:
```
template <typename... types>
static inline std::array<int, sizeof...(types)> make_array (types ... values)
{
    return std::array<int, sizeof...(types)>{values...};
}
```